.travis.yml

```diff
@@ -29,7 +30,9 @@ matrix:
     - gemfile: gemfiles/rails_edge.gemfile
   exclude:
     - rvm: ruby-head
-      gemfile: gemfiles/rails_42.gemfile
+      gemfile:
+        - gemfiles/rails_42.gemfile
```

Comment:
Question: Are these exclusions skipping tests for these combinations because these ruby / rails versions don't work the the library? If so, should I add rails 4.1 and 4.0 to everywhere rails 4.2 is listed?

Reply:
> Are these exclusions skipping tests for these combinations because these ruby / rails versions don't work the the library?

It's more like that each Rails version supports different versions of Ruby. Artemis is supposed to be used on Rails, and if Rails doesn't support a particular version of Ruby, there is no way for Artemis to support the Ruby version Rails doesn't support. See [Rails' `.travis.yml`](https://github.com/rails/rails/blob/9be9597e510d185ca7964d0a05b4ea2a7f2d50d1/.travis.yml#L5-L8) to know what Rubies are supported.

(Rails 4.0 doesn't officially support Ruby 2.3 and 2.4, but I don't mind adding builds for them if they all pass)

> If so, should I add rails 4.1 and 4.0 to everywhere rails 4.2 is listed?

My quick answer is yes, but the logic to determine the build matrix for Rails 4.0 has nothing to do with the build matrix for Rails 4.2.

Reply:
If I got you right, from the travis link, Rails 4.x officially only supports up to Ruby 2.2. So we would exclude them from all newer versions. I'll try up to 2.4 as you mentioned.